### PR TITLE
Update Limbo test suite and fix relevant bugs

### PR DESCRIPTION
**Pull request by randombit**
We did not reject an empty EKU extension, when this is invalid.

We did not check subject alternative name DNS entries at all

### CODE REVIEW

**Comment by randombit** [comment]
Neither of these seem critical so I'm pushing this to 3.7.0 milestone.

File: src/lib/x509/alt_name.cpp [hunk 1]
```diff
@@ -109,6 +109,53 @@ void AlternativeName::encode_into(DER_Encoder& der) const {
...
+   namely A-Z, a-z, 0-9, hypen, and dot, plus '*' for wildcarding.
+   */
+   const uint8_t DNS_TABLE[128] = {
+      0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+      0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+      0x00, 0x00, 0x00, 0x00, 0x2A, 0x00, 0x00, 0x2D, 0x2E, 0x00, 0x30, 0x31, 0x32, 0x33, 0x34, 0x35, 0x36, 0x37, 0x38,
+      0x39, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x61, 0x62, 0x63, 0x64, 0x65, 0x66, 0x67, 0x68, 0x69, 0x6A, 0x6B,
+      0x6C, 0x6D, 0x6E, 0x6F, 0x70, 0x71, 0x72, 0x73, 0x74, 0x75, 0x76, 0x77, 0x78, 0x79, 0x7A, 0x00, 0x00, 0x00, 0x00,
+      0x00, 0x00, 0x61, 0x62, 0x63, 0x64, 0x65, 0x66, 0x67, 0x68, 0x69, 0x6A, 0x6B, 0x6C, 0x6D, 0x6E, 0x6F, 0x70, 0x71,
+      0x72, 0x73, 0x74, 0x75, 0x76, 0x77, 0x78, 0x79, 0x7A, 0x00, 0x00, 0x00, 0x00, 0x00};
```
Comment (reneme) [inline src/lib/x509/alt_name.cpp:138]: Perhaps like that, to make the intent of this visually clearer:

```suggestion
   // clang-format off
   constexpr uint8_t DNS_CHAR_MAPPING[128] = {
      '\0', '\0', '\0', '\0', '\0', '\0', '\0', '\0', '\0', '\0', '\0', '\0', '\0', '\0', '\0', '\0', '\0', '\0', '\0',
      '\0', '\0', '\0', '\0', '\0', '\0', '\0', '\0', '\0', '\0', '\0', '\0', '\0', '\0', '\0', '\0', '\0', '\0', '\0',
      '\0', '\0', '\0', '\0',  '*', '\0', '\0',  '-',  '.', '\0',  '0',  '1',  '2',  '3',  '4',  '5',  '6',  '7',  '8',
       '9', '\0', '\0', '\0', '\0', '\0', '\0', '\0',  'a',  'b',  'c',  'd',  'e',  'f',  'g',  'h',  'i',  'j',  'k',
       'l',  'm',  'n',  'o',  'p',  'q',  'r',  's',  't',  'u',  'v',  'w',  'x',  'y',  'z', '\0', '\0', '\0', '\0',
      '\0', '\0',  'a',  'b',  'c',  'd',  'e',  'f',  'g',  'h',  'i',  'j',  'k',  'l',  'm',  'n',  'o',  'p',  'q',
       'r',  's',  't',  'u',  'v',  'w',  'x',  'y',  'z', '\0', '\0', '\0', '\0', '\0',
   };
   // clang-format on
```

... note that I changed the name of the array and am not sure about the clang-format implication. Perhaps better to copy that in your editor. 😅

File: src/lib/x509/alt_name.cpp [hunk 2]
```diff
@@ -109,6 +109,53 @@ void AlternativeName::encode_into(DER_Encoder& der) const {
...
+
+   std::string canon;
+
+   for(char c : name) {
+      const uint8_t cu = static_cast<uint8_t>(c);
+      if(cu >= 128) {
+         throw Decoding_Error("DNS name should not have any high bits set");
```
Comment (reneme) [inline src/lib/x509/alt_name.cpp:145]:
```suggestion
         throw Decoding_Error("DNS name must not contain any extended ASCII code points");
```

File: src/lib/x509/alt_name.cpp [hunk 3]
```diff
@@ -109,6 +109,53 @@ void AlternativeName::encode_into(DER_Encoder& der) const {
...
+   const uint8_t DNS_TABLE[128] = {
+      0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+      0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+      0x00, 0x00, 0x00, 0x00, 0x2A, 0x00, 0x00, 0x2D, 0x2E, 0x00, 0x30, 0x31, 0x32, 0x33, 0x34, 0x35, 0x36, 0x37, 0x38,
+      0x39, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x61, 0x62, 0x63, 0x64, 0x65, 0x66, 0x67, 0x68, 0x69, 0x6A, 0x6B,
+      0x6C, 0x6D, 0x6E, 0x6F, 0x70, 0x71, 0x72, 0x73, 0x74, 0x75, 0x76, 0x77, 0x78, 0x79, 0x7A, 0x00, 0x00, 0x00, 0x00,
+      0x00, 0x00, 0x61, 0x62, 0x63, 0x64, 0x65, 0x66, 0x67, 0x68, 0x69, 0x6A, 0x6B, 0x6C, 0x6D, 0x6E, 0x6F, 0x70, 0x71,
+      0x72, 0x73, 0x74, 0x75, 0x76, 0x77, 0x78, 0x79, 0x7A, 0x00, 0x00, 0x00, 0x00, 0x00};
+
+   std::string canon;
```
Comment (reneme) [inline src/lib/x509/alt_name.cpp:140]: Perhaps?

```suggestion
   std::string canon;
   conan.reserve(name.size());
```

File: src/scripts/ci/setup_gh_actions.sh [hunk 4]
```diff
@@ -156,7 +156,7 @@ if type -p "apt-get"; then
 
     elif [ "$TARGET" = "limbo" ]; then
         sudo apt-get -qq install python3-dateutil
-        wget -nv https://raw.githubusercontent.com/C2SP/x509-limbo/bd88042508ccfde351b2fee293aebda8971fbebb/limbo.json -O "${SCRIPT_LOCATION}/../../../limbo.json"
+        wget -nv https://raw.githubusercontent.com/C2SP/x509-limbo/f98aa03f45d108ae4e1bc5a61ec4bd0b8d137559/limbo.json -O "${SCRIPT_LOCATION}/../../../limbo.json"
```
Comment (reneme) [inline src/scripts/ci/setup_gh_actions.sh:159]: Perhaps this could live in a future central repo config (see #4390)

**Review by reneme — APPROVED** [review]
Just some nits

**Pull request merged**